homeassistant/components/roon/media_player.py

```diff
@@ -52,12 +55,38 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+SERVICE_SYNC = "sync"
+SERVICE_UNSYNC = "unsync"
+SERVICE_TRANSFER = "transfer"
+
+ATTR_LINK_NAME = "link_name"
+ATTR_UNLINK_NAME = "unlink_name"
+ATTR_TRANSFER_NAME = "transfer_name"
+
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up Roon MediaPlayer from Config Entry."""
     roon_server = hass.data[DOMAIN][config_entry.entry_id]
     media_players = set()
 
+    # Register entity services
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_SYNC,
+        {vol.Required(ATTR_LINK_NAME): cv.string},
```

Comment:
I think that it would make more sense to use entity IDs for this, as that's the identifier the user knows.

Reply:
Also, should it allow to link multiple at once? For Sonos, the service call takes which player leads and what others are following.

Reply:
The nice thing about the way Sonos does it is that it's a declarative interface, you have a service call that allows users to specify desired state, and the integration figures it out. This service is imperative in that the user needs to make the steps of ungrouping/grouping the right players to get to the desired state. That's cumbersome.

Reply:
I've reworked along the lines you suggested, changing to use `entity_ids`, using lists, and adding the ability to unjoin everything easily.

Join isn't quite declarative (unless you call unjoin with no parameters first) - but this allows more flexibility (eg for adding one zone to the existing ones without needing to know what is already joined - which is how I plan to use it).